tls/s2n_connection.h

```diff
@@ -40,6 +40,14 @@
 
 #define is_handshake_complete(conn) (APPLICATION_DATA == s2n_conn_get_current_message_type(conn))
 
+typedef enum {
+    S2N_NO_TICKET = 0,
+    S2N_EXPECTING_NEW_TICKET,
+    S2N_ATTEMPT_DECRYPT_TICKET,
+    S2N_RENEW_TICKET,
+    S2N_RECEIVED_VALID_TICKET
+} s2n_session_ticket_status;
```

Comment:
Is there a way of unifying this status with the handshake state machine states? I personally am having a hard time tracing the session ticket status along with the handshake states. Could we get away with just doing `conn->handshake.handshake_type |= WITH_SESSION_TICKET` in strategic places?